src/admin.rs

```diff
@@ -18,85 +27,71 @@ use node_bindgen::core::JSValue;
         Ok(new_instance)
     }
 }
 
-
-pub struct AdminScClient {
-    inner: Option<SharedScClient>,
+pub struct FluvioAdminJS {
+    inner: Option<SharedFluvio>,
 }
 
 #[node_bindgen]
-impl AdminScClient {
+impl FluvioAdminJS {
     #[node_bindgen(constructor)]
     pub fn new() -> Self {
         Self { inner: None }
     }
 
-
-    pub fn set_client(&mut self, client: SharedScClient) {
+    pub fn set_client(&mut self, client: SharedFluvio) {
         self.inner.replace(client);
     }
 
-    
     #[node_bindgen]
-    async fn list_topic(&self) -> Result<ArrayBuffer,ClientError> {
-
-        debug!("list topics");
-
-        let client = self.inner.as_ref().unwrap().clone();
-        let mut client_w = client.write().await;
-
-        let list_topics = client_w.topic_metadata(None).await?;
-
-        let json_slice = serde_json::to_vec(&list_topics)
-        .map_err(|err| ClientError::Other(format!("serialization error: {}",err.to_string())))?;
-
-        // convert to array buffer and wrap in the buffer
-        Ok(ArrayBuffer::new(json_slice))
-
+    async fn list_topic(&self) -> Result<ArrayBuffer, FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            let topics = client.list::<TopicSpec, _>(vec![]).await?;
+
+            let json_slice = serde_json::to_vec(&topics).map_err(|err| {
+                FluvioError::Other(format!("serialization error: {}", err.to_string()))
+            })?;
+            // // convert to array buffer and wrap in the buffer
+            Ok(ArrayBuffer::new(json_slice))
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
     }
 
     #[node_bindgen]
-    async fn find_topic(&self,topic_name: String) -> Result<TopicInfo,ClientError> {
-
-        debug!("trying find topic: {}",topic_name);
+    async fn find_topic(&self, topic_name: String) -> Result<TopicInfo, FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            let topics = client.list::<TopicSpec, _>(vec![]).await?;
 
-        let client = self.inner.as_ref().unwrap().clone();
-        let mut client_w = client.write().await;
+            let topic = topics.iter().find(|topic| topic.name == topic_name);
 
-        let list_topics = client_w.topic_metadata(None).await?;
-
-        debug!("topics: {:#?}",list_topics);
-        if let Some(topic) = list_topics.iter().find( |topic| topic.name == topic_name) {
-            debug!("found topic {}",topic_name);
-            let json = serde_json::to_vec(topic)
-                .map_err(|err| ClientError::Other(format!("serialization error: {}",err.to_string())))?;
-                
+            let json = serde_json::to_vec(&topic).map_err(|err| {
+                FluvioError::Other(format!("serialization error: {}", err.to_string()))
```

Comment:
We can make an issue for this rather than worrying about it here, but we should make specialized error types for these, such as `FluvioError::Json`

Reply:
Json parsing is not concern of Fluvio